evm_arithmetization/src/cpu/kernel/optimizer.rs

```diff
@@ -157,6 +159,48 @@ fn remove_ignored_values(code: &mut Vec<Item>) {
     });
 }
 
+/// First law: (not A) and (not B) = not (A or B)
+/// [PUSH a, NOT, PUSH b, NOT, AND] -> [PUSH a, PUSH b, OR, NOT]
```

Comment:
I think we'd want to catch any DUPs as well instead of limiting it to PUSHes

Reply:
Should work now.